compiler/rustc_typeck/src/impl_wf_check.rs

```diff
@@ -207,17 +207,25 @@ fn enforce_impl_params_are_constrained(
 }
 
 fn report_unused_parameter(tcx: TyCtxt<'_>, span: Span, kind: &str, name: &str) {
-    struct_span_err!(
+    let mut err = struct_span_err!(
         tcx.sess,
         span,
         E0207,
         "the {} parameter `{}` is not constrained by the \
         impl trait, self type, or predicates",
         kind,
         name
-    )
-    .span_label(span, format!("unconstrained {} parameter", kind))
-    .emit();
+    );
+    err.span_label(span, format!("unconstrained {} parameter", kind));
+    if kind == "const" {
+        err.note(
+            "expressions using a const parameter must map each value to a distinct output value",
+        );
+        err.note(
+            "proving the result of expressions other than the parameter are unique is not supported",
```

Comment:
I think the meaning of this comment may not be entirely clear to someone unfamiliar with the constraints of const generics, but I'm not sure of better wording yet. I'll have a think about it.